include/dem/particle_particle_contact_force.h

```diff
@@ -1129,131 +1045,283 @@ class ParticleParticleContactForce
-    double normal_force_norm =
-      normal_spring_constant * normal_overlap +
-      normal_damping_constant * normal_relative_velocity_value;
-
-    // Calculation of tangential force. Since we need damping tangential force
-    // in the gross sliding again, we define it as a separate variable
-    Tensor<1, 3> damping_tangential_force =
-      tangential_damping_constant * tangential_relative_velocity;
-    tangential_force =
-      (tangential_spring_constant * contact_info.tangential_overlap) +
-      damping_tangential_force;
+    const double F_so = 6.283185307179586 * // 2. * M_PI
+                        this->effective_radius *
+                        this->effective_surface_energy.at(property_index);
 
-    double coulomb_threshold =
-      this->effective_coefficient_of_friction[vec_particle_type_index(
-        particle_one_type, particle_two_type)] *
-      normal_force_norm;
+    const double delta_0 = -std::sqrt(
+      this->effective_hamaker_constant.at(property_index) / (6. * F_so));
 
-    // Check for gross sliding
-    const double tangential_force_norm = tangential_force.norm();
-    if (tangential_force_norm > coulomb_threshold)
+    // Cohesive force. This will need to be added to the
+    // first vector inside the tuple.
+    double cohesive_term;
+    if (normal_overlap > delta_0)
       {
-        // Gross sliding occurs and the tangential overlap and tangential
-        // force are limited to Coulomb's criterion
-        contact_info.tangential_overlap =
-          (coulomb_threshold *
-             (tangential_force / (tangential_force_norm + DBL_MIN)) -
-           damping_tangential_force) /
-          (tangential_spring_constant + DBL_MIN);
+        cohesive_term = -F_so;
+
+        if (normal_overlap > 0.)
+          {
+            this->template calculate_hertz_mindlin_limit_overlap_contact<
+              Parameters::Lagrangian::ParticleParticleContactForceModel::DMT>(
+              contact_info,
+              tangential_relative_velocity,
+              normal_relative_velocity_value,
+              normal_unit_vector,
+              normal_overlap,
+              particle_one_properties,
+              particle_two_properties,
+              normal_force,
+              tangential_force,
+              particle_one_tangential_torque,
+              particle_two_tangential_torque,
+              rolling_resistance_torque);
+          }
+        else
+          {
+            contact_info.tangential_overlap = Tensor<1, 3>();
+          }
+      }
+    else
+      {
+        cohesive_term = -this->effective_hamaker_constant.at(property_index) *
+                        effective_radius /
+                        (6. * Utilities::fixed_power<2>(normal_overlap));
 
-        tangential_force =
-          (tangential_spring_constant * contact_info.tangential_overlap) +
-          damping_tangential_force;
+        contact_info.tangential_overlap = Tensor<1, 3>();
```

Comment:
Ok, I'm confused by the different cases with the nested ifs. I suggest a more straightforward like that: 


```
if (normal_overlap > 0.)
      {
	// Case for contact between particles

	// Contact forces contribution
        this->template calculate_hertz_mindlin_limit_overlap_contact<
          Parameters::Lagrangian::ParticleParticleContactForceModel::DMT>(
          contact_info,
          tangential_relative_velocity,
          normal_relative_velocity_value,
          normal_unit_vector,
          normal_overlap,
          particle_one_properties,
          particle_two_properties,
          normal_force,
          tangential_force,
          particle_one_tangential_torque,
          particle_two_tangential_torque,
          rolling_resistance_torque);

	// Cohesive contribution
	cohesive_term = -F_so;
	normal_force += cohesive_term * normal_unit_vector;
      }
else if (normal_overlap > delta_0)
  {
	// Case for particle within the ... range

	// Only long range cohesive contribution
	normal_force -= F_so * normal_unit_vector;

	// Reset of the tangential overlap history
        contact_info.tangential_overlap = Tensor<1, 3>();
  }
else
  {
        // Case ...

	// Only long range cohesive contribution
        double cohesive_term = -this->effective_hamaker_constant.at(property_index) *
                    effective_radius /
                    (6. * Utilities::fixed_power<2>(normal_overlap));

        normal_force += cohesive_term * normal_unit_vector;

        // Reset of the tangential overlap history
        contact_info.tangential_overlap = Tensor<1, 3>();
  }

```
(this is an example, but it should looks like that)
Because as the none implementor I do know really understand the cases and nested ifs are kinda error-prone.

Reply:
It applies also for p-w

Reply:
And forget about my previous comment

Reply:
yeah it's kinda weird this embedded if. Are you sure it's necessaryt o have two if/else like that?

Reply:
Its either Audrey's implementation or mine.  (both are fine by me)

There is 3 cases. (Sorry, the background is translucent)
![image](https://github.com/user-attachments/assets/47c54e44-193c-4814-99ff-adcb3404c925)

When the overlap is positive (particles are touching) there is the contact force + a constant cohesive force (doesn't vary relative to the overlap). 

When the overlap is a little bit negative, there is no contact force but there still is the same constant cohesive force. 

Lastly, when the overlap is more negative, There is a cohesive force that depends (vary) on the overlap.